compiler/rustc_session/src/config.rs

```diff
@@ -2451,6 +2451,19 @@ pub fn parse_externs(
             Some((opts, name)) => (Some(opts), name.to_string()),
         };
 
+        if !rustc_lexer::is_ident(&name) {
+            let mut error = handler.early_struct_error(format!(
+                "crate name `{name}` passed to `--extern` is not a valid identifier"
```

Comment:
I briefly considered `Debug`-printing `name` instead (via `{:?}` or via `.escape_debug()`) to escape line breaks but it looked weird.